shared-services/src/files/aws-s3-file.service.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from "@nestjs/common"
+import { AwsS3FileUploader } from "./aws-s3-file-uploader"
+import { AwsS3FileServiceConfig } from "./file-service-config"
+import { FileServiceInterface } from "./file-service.interface"
+import { S3Client } from "@aws-sdk/client-s3"
+
+@Injectable()
+export class AwsS3FileService implements FileServiceInterface {
+  constructor(
+    private readonly awsS3FileUploader: AwsS3FileUploader,
+    private readonly awsS3FileServiceConfig: AwsS3FileServiceConfig,
+    private readonly s3Client: S3Client
+  ) {}
+
+  async putFile(
+    key: string,
+    file: File,
+    setProgressValue: (value: number) => void
+  ): Promise<string> {
+    const id = await this.awsS3FileUploader.uploadAwsS3File(
+      this.s3Client,
+      file,
+      setProgressValue,
+      this.awsS3FileServiceConfig.bucketName
+    )
+    return id
+  }
+  getDownloadUrlForPhoto(id: string): string {
+    return `https://${this.awsS3FileServiceConfig.bucketName}.s3.${this.awsS3FileServiceConfig.region}.amazonaws.com/${id}`
+    /*const getObjectCommand = new GetObjectCommand({
```

Comment:
I tried using the getSignedUrl method to get the download URL but that caused issues because it returns a Promise<string> instead of a string. All existing clients expect a string and when I tried changing them into async methods that await on the Promise<string>, I ran into a bunch of react issues of this type https://stackoverflow.com/questions/47658765/objects-are-not-valid-as-a-react-child-found-object-promise. I decided to just create the URL directly since it's fairly straightforward and is also already done this way for Cloudinary download URLs.

Reply:
Removing this commented out code before merging.